packages/webex/src/calling.js

```diff
@@ -68,6 +70,67 @@ class Calling extends EventEmitter {
       });
   }
 
+  async deregister() {
+    if (!this.registered) {
+      this.log.info('Authentication: webex.internal.device.deregister already done', logContext);
+
+      return Promise.resolve();
+    }
+
+    const promises = [];
+
+    const lineIds = Object.keys(this.callingClient?.getLines());
+
+    const lines = lineIds.length ? this.callingClient.getLines() : {};
+    for (const lineId of lineIds) {
+      const line = lines[lineId];
+      if (line.getStatus() === 'active') {
+        const thisLinePromise = new Promise((resolve, reject) => {
```

Comment:
Because the line deregister can be confirmed as success only after receiving the success event. Therefore, instead of asking the user to wait for deregister of all the lines, we wait for event and resolve it. 

When we do this the `await Calling.deregister()` will wait until all lines are deregistered, mercury is disconnected and WDM device is deleted

Reply:
It just occurs to me, even here, we can assume that the user doesn't care about results anymore. Right?

@sreenara

Reply:
Yes. We don't have to care about preserving status of response and handling it anywhere because this is the end of the workflow. @mkesavan13

Reply:
Removed the promise mechanism as well